src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -611,13 +611,17 @@ string ReserveName(string prefix)
                 return count == 0 ? prefix : $"{prefix}{count}";
             }
 
+            // Helper to emit a label.  As of C# 10, labels aren't statements of their own and need to adorn a following statement;
+            // if a label appears just before a closing brace, then, it's a compilation error.  To avoid issues there, this by
+            // default implements a blank statement (a semicolon) after each label, but individual uses can opt-out of the semicolon
+            // when it's known the label will always be followed by a statement.
```

Comment:
NIT: perhaps move this and ReserveNames after return statement? Not sure if it is part of the coding guidelines or not, but if it is we could also consider in the future adding an analyzer to ensure localfuncs are after a return statement.